packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts

```diff
@@ -115,48 +107,50 @@ export class GriffelCSSExtractionPlugin {
+        name: 'griffel',
+        test: isGriffelCSSModule,
+        chunks: 'all',
+        enforce: true,
+      };
+    }
+
     compiler.hooks.compilation.tap(PLUGIN_NAME, compilation => {
-      // A chunk where we will move all CSS modules
-      const griffelChunk = compilation.addChunk('griffel');
-      // Set up a dummy module to prevent the chunk from getting cleaned up by RemoveEmptyChunksPlugin
-      const dummyModule = new GriffelDummyModule();
+      if (!compiler.options.optimization.splitChunks) {
+        // Performs module movements if SplitChunksPlugin is not present
+        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: OPTIMIZE_CHUNKS_STAGE_ADVANCED }, () => {
+          moveCSSModulesToGriffelChunk(compilation);
+        });
+      }
 
+      // In this case we detach Griffel chunk from all other chunks and connect it with the main entrypoint
       if (this.attachToMainEntryPoint) {
-        compilation.hooks.afterChunks.tap(PLUGIN_NAME, () => {
+        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: 10 }, () => {
+          const griffelChunk = compilation.namedChunks.get('griffel');
+
+          if (typeof griffelChunk === 'undefined') {
+            return;
+          }
+
+          griffelChunk.disconnectFromGroups();
           attachGriffelChunkToMainEntryPoint(compilation, griffelChunk);
         });
       }
 
-      // Adds dummy module here to try to make sure that other module optimization steps won't conflict
-      compilation.hooks.afterOptimizeModules.tap(PLUGIN_NAME, () => {
-        compilation.modules.add(dummyModule);
-        compilation.chunkGraph.connectChunkAndModule(griffelChunk, dummyModule);
-      });
-
-      // Performs movements before SplitChunksPlugin
-      compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: OPTIMIZE_CHUNKS_STAGE_BASIC }, () => {
-        moveCSSModulesToGriffelChunk(compilation, this.attachToMainEntryPoint);
-      });
-
-      compilation.hooks.afterOptimizeChunks.tap(PLUGIN_NAME, () => {
-        // If SplitChunksPlugin has configured "cacheGroups", we need to move chunks again to ensure that needed
-        // modules are still attached "griffel" chunk
-        if (compiler.options.optimization.splitChunks && compiler.options.optimization.splitChunks.cacheGroups) {
-          moveCSSModulesToGriffelChunk(compilation, this.attachToMainEntryPoint);
-        }
-
-        // Remove dummy module once we are sure chunk optimization steps have finished. i.e. won't conflict with
-        // SplitChunksPlugin
-        compilation.modules.delete(dummyModule);
-        compilation.chunkGraph.disconnectChunkAndModule(griffelChunk, dummyModule);
-      });
-
       compilation.hooks.processAssets.tap(
         {
           name: PLUGIN_NAME,
           stage: Compilation.PROCESS_ASSETS_STAGE_PRE_PROCESS,
         },
         assets => {
+          const griffelChunk = compilation.namedChunks.get('griffel');
```

Comment:
is the check for CSS assets under this new early return still necessary? there should always be assets if a griffel chunk exists?

Reply:
I would keep it to make TS happy and `.find()` can return `undefined`